pyfr/backends/cuda/blasext.py

```diff
@@ -62,33 +62,47 @@ def errest(self, x, y, z, *, norm):
         grid = get_grid_for_block(block, ncolb, ncola)
 
         # Empty result buffer on the device
-        err_dev = cuda.mem_alloc(ncola*grid[0]*x.itemsize)
+        reduced_dev = cuda.mem_alloc(ncola*grid[0]*rs[0].itemsize)
 
         # Empty result buffer on the host
-        err_host = cuda.pagelocked_empty((ncola, grid[0]), dtype)
+        reduced_host = cuda.pagelocked_empty((ncola, grid[0]), dtype)
+
+        tplargs = dict(norm=norm, sharesz=block[0], method=method)
+
+        if method == 'resid':
+            dt = dt_mat if dt_mat else dt_fac
+
+            tplargs['dt_type'] = 'matrix' if dt_mat else 'scalar'
```

Comment:
Have dt_fac default to one and pass through

Reply:
dt_fac is assigned to _dtau when there is such attribute available just before the reduction kernel call;
https://github.com/semi-h/PyFR/blob/rkvdh2-pseudo/pyfr/integrators/base.py#L179
and with pseudo controller there is always a _dtau;
https://github.com/semi-h/PyFR/blob/rkvdh2-pseudo/pyfr/integrators/dual/pseudo/base.py#L25
so dt_fac is always assigned to a value when dual time stepping is used.

If I have dt_fac default to one this will only apply when errest kernel is used, and then it won't have an effect at the end.

Reply:
Passing dtau is not the best way I think. Maybe we can find a better solution if you explain a possible use case for this. For example how about something like an alpha parameter that we can supply at runtime? Then it'll be easy and less complicated.